Extension/src/common.ts

```diff
@@ -89,7 +89,7 @@ export async function getRawJson(path: string | undefined): Promise<any> {
 
 export function fileIsCOrCppSource(file: string): boolean {
     const fileExtLower: string = path.extname(file).toLowerCase();
-    return ["cu", ".C", ".c", ".cpp", ".cc", ".cxx", ".mm", ".ino", ".inl"].some(ext => fileExtLower === ext);
+    return [".cu", ".c", ".cpp", ".cc", ".cxx", ".c++", ".cp", ".tcc", ".mm", ".ino", ".ipp", ".inl"].some(ext => fileExtLower === ext);
```

Comment:
There may be some more code here that needs updating.  The ".C" being upper case was intentional, as only uppercase ".C" is associated with C++ by default.

Reply:
I don't think so -- the input file is made lowercased and the function name is COrCpp. I don't see any C++ specific code that uses this function.